test_opensearchpy/test_connection.py

```diff
@@ -502,6 +504,46 @@ def urlopen_raise(*_, **__):
         assert str(e.value) == "Wasn't modified!"
 
 
+class TestAsyncHttpConnection(TestCase):
```

Comment:
Let's pull this into a separate file? Either make an `async` folder with a bunch of tests or `test_async_http_connection.py`?

Reply:
Ok. Sure. I wasn't sure where to put it since it looked like this class didn't have a single test to its name :)